### PR TITLE
feat(vscode): Support open chat in editor / open chat in new window …

### LINKED ISSUE #2944: Support open chat in editor / open chat in new window for Tabby Chat
**Please describe the feature you want**

image

GitHub Copilot supports opening a `Chat` as an individual window or as a separate tab, which is also a feature that can be implemented for Tabby Chat functionality.

**Additional context**
1. https://demo.tabbyml.com/files/github/TabbyML/tabby/-/blob/77515f9c83f32e0db84d454cad671e4a90a032e7/clients/vscode/package.json#L402
2. https://code.visualstudio.com/api/extension-guides/webview (createWebviewPanel)

---
Please reply with a 👍 if you want this feature.

### PR DESCRIPTION
**Pull request by antimonyGu**
close #2944

- Test:

Manually test create chat panel then send a hello world message.

- Recording:
https://jam.dev/c/606f0750-4c5c-44e5-9c4a-8c2ce06ddf2f cc @icycodes

### CODE REVIEW

**Comment by antimonyGu** [comment]
`getFileContextFromSelection` `createClient` `getFileContextFromEditor` to be refactored into helper cause they couple with each other. @wsxiaoys

**Comment by antimonyGu** [comment]
rebase version of PR(https://github.com/TabbyML/tabby/pull/3131). cc @wsxiaoys

**Comment by antimonyGu** [comment]
@icycodes Can you please review this PR when you are free? Thanks!

**Comment by antimonyGu** [comment]
@icycodes BTW, Can you please re-trigger the CI jobs. Thansk! cc @wsxiaoys

File: clients/vscode/package.json [hunk 1]
```diff
@@ -139,6 +139,11 @@
         "command": "tabby.server.selectPastServerConfig",
         "title": "Select Server Endpoint from History",
         "category": "Tabby"
+      },
+      {
+        "command": "tabby.chat.createPanel",
+        "title": "Create new panel",
```
Comment (wsxiaoys) [inline clients/vscode/package.json:145]:
```suggestion
        "title": "New Chat",
```

File: clients/vscode/src/Commands.ts [hunk 2]
```diff
@@ -276,6 +279,11 @@ export class Commands {
     "chat.generateCodeBlockTest": async () => {
       this.sendMessageToChatPanel("Generate a unit test for the selected code:");
     },
+    "chat.createPanel": async () => {
+      const panel = window.createWebviewPanel("tabby.chatView", "Tabby Chat", ViewColumn.One, {});
```
Comment (wsxiaoys) [inline clients/vscode/src/Commands.ts:283]:
```suggestion
      const panel = window.createWebviewPanel("tabby.chatView", "Tabby", ViewColumn.One, {});
```

**Review by wsxiaoys — APPROVED** [review]

**Pull request merged**